ethcore/blockchain/src/blockchain.rs

```diff
@@ -988,7 +989,9 @@ impl BlockChain {
 	pub fn epoch_transition_for(&self, parent_hash: H256) -> Option<EpochTransition> {
 		// slow path: loop back block by block
 		for hash in self.ancestry_iter(parent_hash)? {
+			trace!(target: "blockchain", "Got parent hash {} from ancestry_iter", hash);
 			let details = self.block_details(&hash)?;
+			trace!(target: "blockchain", "Block #{}: Got block details", details.number);
```

Comment:
I'd print the `hash` in this log entry as well - it's easier to analyse the logs if you can just filter out one message and don't need the context (i.e. the line above with a hash in this case)